flink-ml-iteration/src/main/java/org/apache/flink/ml/iteration/DataStreamList.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.iteration;
+
+import org.apache.flink.streaming.api.datastream.DataStream;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/** An utility class to maintain a list of {@link DataStream}, which might have different types. */
+public class DataStreamList {
+    public static DataStreamList of(DataStream<?>... streams) {
+        return new DataStreamList(Arrays.asList(streams));
```

Comment:
Hmm... It seems that this static method is added to the FLIP-176 since our last discussion of the API section0.

I am not sure we need this static method given that we already have `public DataStreamList(List<DataStream<?>> dataStreams)`. Could you help explain the motivation for this method?

Reply:
For users to pass the parameters and return IterationBodyResult, they need to construct the `DataStreamList` with several named datastreams. The utility avoids they have to write codes like

```
new DataStreamList(Arrays.asList(datastream1, datastream2));
```